src/scienceDatabase.cpp

```diff
@@ -210,6 +229,30 @@ P<ScienceDatabase> ScienceDatabase::queryScienceDatabase(string name, int32_t pa
     return nullptr;
 }
 
+static int queryScienceDatabaseById(lua_State* L)
+{
+    P<ScienceDatabase> entry = nullptr;
+
+    for(int i = 1; i <= lua_gettop(L); i++)
```

Comment:
This looks off. The lua_gettop is to iterate over all the given parameters, but effectively this then only uses the last one.

Reply:
Changed to check for a number in the first argument instead.